### ISSUE TITLE
XSearch Runtime

### ISSUE THREAD

**Issue by agrier04** [ndaniels/cablastp2#7]
I've downloaded, compiled, and installed cablastp2 using Go without issue.  I compressed the UniRef50 protein sequence database using default parameters, except I specified a shorter minimum match length:

cablastp-compress --makeblastdb="/software/blast/2.2.30+/bin/makeblastdb" --max-seeds="200" --min-match-len="10" --p="24" --ext-seed-size 0 --match-seq-id-threshold 70 --ext-seq-id-threshold 60 ./capblastp_uniref ./uniref_full/uniref50.fasta

This appears to have worked fine - no errors or warnings.  The "compressed" file in the resulting directory is ~6.6Gb; significantly smaller than the pre-compressed NR distribution provided here.

My issue is runtime.  For example, I have a test set of 92 150bp reads.  Blastx tasked to blastx-fast can finish querying the test set against the UniRef50 database in 5-10 minutes with the following parameters:

blastx -query test.fa -task blastx-fast -db ./uniref_full/uniref50 -out blastx_test.tsv -evalue 1.0 -outfmt 6 -max_target_seqs 20 -num_threads 24

The following capblastq-xsearch command has been running on 24 cores with 64Gb of memory for 11 hours:

cablastp-xsearch --blastn="/software/blast/2.2.30+/bin/blastn" --blastx="/software/blast/2.2.30+/bin/blastx" --coarse-eval="1000" --compress-query="true" --makeblastdb="/software/blast/2.2.30+/bin/makeblastdb" --min-match-len="10" --short-queries="true" --p="24" ./capblastp_uniref test.fa --blast-args -evalue 1.0 -outfmt 6 -max_target_seqs 20 -task blastx-fast > capblastp_test.tsv 

Based on the length of the log file, I suspect that it is about half way done.

How can I achieve the performance described in the Yu et. al paper, or even something within couple a orders of magnitude?

**Comment by ndaniels** [comment]
First of all, the minimum match length of 10 seems much too short. I would worry that this will lead to excessive cross-linking of many short subsequences, leading to both missed hits for the coarse search and a great deal of time being spent decompressing results for fine search (which it seems like you are seeing). 

If that doesn't resolve the issue, I would see if the problem is due to query-side compression; you might try not using the --compress-query="true" flag to see if that makes a difference. If it is faster than blastx, then the parameters for query-side compression may need to be adjusted; the min-match-length for compression should be close to the read length (perhaps 120 if you have 150bp reads). That said, if there isn't significant coverage depth in your test set (and I doubt there is if there are only 92 reads!) you should really not use query-side clustering, as it will only slow things down.

So to summarize, I would use a longer minimum match length (the default of 40 is reasonable) and not use query-side clustering with that dataset.
